# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1250623735 14400
#      Tue Aug 18 15:28:55 2009 -0400
# Node ID 1d51ab34b509346be849374d3ccc664ed40470c5
# Parent  f5ec5dc668249e2a6f332814a3714eed07947c3a
tweaks for HDF5 library check

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,15 +1,22 @@
+2009-08-18  John W. Eaton  <jwe@octave.org>
+
+	* aclocal.m4 (OCTAVE_HDF5_HAS_REQUIRED_API): If test fails, set
+	octave_hdf5_ok to no instead of setting HAVE_HDF5 to false.
+	* configure.in: Check octave_hdf5_ok instead of HAVE_HDF5.
+	Set TEXINFO_HDF5 if HDF5 library is found.
+
 2009-08-18  John W. Eaton  <jwe@octave.org>
 
 	* Makeconf.in (ALL_CPPFLAGS): Add HDF5_CPPFLAGS to the list.
 	* aclocal.m4 (OCTAVE_HDF5_HAS_REQUIRED_API):
 	Set HAVE_HDF5 to false, not WITH_HDF5.
 	* configure.in: Use OCTAVE_CHECK_LIBRARY to check for HDF5
-	library.  Print HDF5_CPPFLAGS and HDF5_LDFLAGLS in the summary.
+	library.  Print HDF5_CPPFLAGS and HDF5_LDFLAGS in the summary.
 	* octave-bug.in, octave-bug.cc.in, Makeconf.in: Substitute
 	HDF5_CPPFLAGS and HDF5_LDFLAGS.
 
 2009-08-18  John W. Eaton  <jwe@octave.org>
 
 	* acx_blas_f77_func.m4 (ACX_BLAS_F77_FUNC): Save and restore LIBS.
 
 2009-08-18  John W. Eaton  <jwe@octave.org>
diff --git a/aclocal.m4 b/aclocal.m4
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1007,17 +1007,17 @@ AC_DEFUN([OCTAVE_HDF5_HAS_REQUIRED_API],
     AC_TRY_LINK([
 #define H5_USE_16_API 1
 #include <hdf5.h>
 ], [
   H5Eset_auto (0, 0);], [
       octave_cv_hdf5_has_required_api=yes], [
       octave_cv_hdf5_has_required_api=no])])
   if test "$octave_cv_hdf5_has_required_api" = "no"; then
-    HAVE_HDF5=false
+    octave_hdf5_ok=no
     warn_hdf5="HDF5 library does not provide the version 1.6 API.  Octave will not be able to save or load HDF5 data files."
     AC_MSG_WARN($warn_hdf5)
   fi
 ])
 dnl
 dnl Check for the QHull version.
 dnl
 AC_DEFUN(OCTAVE_CHECK_QHULL_VERSION,
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -500,17 +500,18 @@ save_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$Z_CPPFLAGS $CPPFLAGS"
 save_LIBS="$LIBS"
 LIBS="$Z_LDFLAGS $Z_LIBS $LIBS"
 OCTAVE_CHECK_LIBRARY(hdf5, HDF5,
   [HDF5 library not found.  Octave will not be able to save or load HDF5 data files.],
   [hdf5.h], [H5Gget_num_objs], [], [],
   [warn_hdf5=
    OCTAVE_HDF5_HAS_REQUIRED_API
-   if test $HAVE_HDF5; then
+   if test $octave_hdf5_ok = yes; then
+     TEXINFO_HDF5="@set HAVE_HDF5"
      AC_DEFINE(HAVE_HDF5, 1, [Define if HDF5 is available.])
      AC_DEFINE(HAVE_H5GGET_NUM_OBJS, 1, [Define if HDF5 has H5Gget_num_objs.])
      case "$canonical_host_type" in
        *-*-msdosmsvc)
 	 OCTAVE_HDF5_DLL
 	 ;;
      esac
    fi])
